### PR TITLE
[deckhouse-controller] add sdk

### PR DESCRIPTION
**Pull request by ldmonster**
## Description

Add batch hook in addon operator and integrate it to deckhouse infrastructure
Add log field Err to easy error handling
If stacktrace in log will be not JSON - escape, if not - nested
DHCTL - remove addon operator dependencies, add Module-SDK dependencies

## Why do we need it, and what problem does it solve?

New feature to use batch hook (binary hook with module-sdk spec)
Logger impovements
DHCTL has less dependencies and more flexible

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller, dhctl, pkg
type: feat
summary: add batch hook in addon operator, minor refactoring, bump dependencies
impact_level: low
```

### CODE REVIEW

File: deckhouse-controller/pkg/controller/module-controllers/release/controller.go [hunk 1]
```diff
@@ -956,7 +958,7 @@ func validateModule(def moduleloader.Definition, values addonutils.Values, logge
 	if err != nil {
 		return fmt.Errorf("read open API files: %w", err)
 	}
-	dm, err := addonmodules.NewBasicModule(def.Name, def.Path, def.Weight, nil, cb, vb, logger.Named("basic-module"))
+	dm, err := addonmodules.NewBasicModule(def.Name, def.Path, def.Weight, nil, cb, vb, app.CRDsFilters, shapp.DebugKeepTmpFiles == "yes", logger.Named("basic-module"))
```
Comment (yalosev) [inline deckhouse-controller/pkg/controller/module-controllers/release/controller.go:961]:
```suggestion
	dm, err := addonmodules.NewBasicModule(def.Name, def.Path, def.Weight, nil, cb, vb, app.CRDsFilters, shapp.DebugKeepTmpFiles == "yes", logger.Named("basic-module" ))
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by yalosev** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 12160217902 12160217902-1-con-1-27 /sys/deckhouse-oss/install:pr10617 ec2-user@35.156.5.101

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/openstack'...

**Comment by yalosev** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.29 12161603008 12161603008-1-con-1-29 /sys/deckhouse-oss/install:pr10617 ec2-user@3.125.186.54

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've recognized your '/e2e/abort' command and started the workflow...

**Review by yalosev — APPROVED** [review]

**Pull request merged**